### ISSUE TITLE
kickasstorrent search error trough API

### ISSUE THREAD

**Issue by muhahacz** [Jackett/Jackett#1049]
Docker file from LinuxServer.io, Jackett Version 0.7.1006.0

Request
> http://192.168.0.251:9876/potato/kickasstorrent?imdbid=tt2404435&search=The+Magnificent+Seven+2016&user=&passkey=rn353aaizbs35dzavyurm1sqcla75ura

Response
```
{
"results": [],
"total_results": 0
}

```

Log
```
CardigannIndexer (kickasstorrent): Error while parsing row '<tr class="t-row"> <td class="ttable_col1" nowrap="nowrap">     <div class="iaconbox center floatright">         <a rel="41096" class="icommentjs kaButton smallButton rightButton" href="comments.php?type=torrent&amp;id=41096">1<i class="ka ka-comment">				</i>
			</a>          <a class="icon16" href="torrents-details.php?id=41096&amp;hit=1" title="Verified Torrent">
				<i class="ka ka16 ka-verify ka-green">				</i>
			</a>                    <a data-download="41096" title="Download torrent file" class="icon16" href="download.php?id=41096&amp;name=The%20Magnificent%20Seven%20%282016%29.iso.torrent">
				<i class="ka ka16 ka-arrow-down">				</i>
			</a>     </div>     <div class="torrentname">          <div class="markeredBlock torType filmType">             <a href="torrents-details.php?id=41096&amp;hit=1" class="cellMainLink">The Magnificent Seven (2016) DVD9 RETAIL NO CINAVIA</a>                  <span class="font11px lightgrey block">                 Posted by <i title="Elite Uploader" class="ka ka-star ka-vul2" style="font-size: 16px;color:orange;">					</i>&nbsp;<a class="plain" href="account-details.php?id=1695">jmanwf</a> in <span id="cat_12155145">
						<strong>
							<a href="torrents.php?parent_cat=Movies">Movies</a>  » DVD ISO                 </strong>
					</span>                           </span>         </div>     </div> </td>
	<td class="ttable_col2" align="center">6.67 GB</td> <td class="ttable_col1" align="center">10-02-2017 09:40:25</td>
	<td class="ttable_col2" align="center">
		<font color="green">1</font>
	</td> <td class="ttable_col1" align="center">
		<font color="#ff0000">0</font>
	</td> </tr>':

System.Exception: Error while parsing field=category, selector=span[id^="cat_"], value=<null>: None of the case selectors "[:has(a:contains("Anime")):contains("English Translated"), 118],[:has(a:contains("Anime")):contains("Other"), 133],[:has(a:contains("Applications")):contains("Handheld"), 144],[:has(a:contains("Applications")):contains("Linux"), 142],[:has(a:contains("Applications")):contains("Mac"), 140],[:has(a:contains("Applications")):contains("Other"), 131],[:has(a:contains("Applications")):contains("Windows"), 139],[:has(a:contains("Books")):contains("Audio books"), 109],[:has(a:contains("Books")):contains("Biography"), 110],[:has(a:contains("Books")):contains("Children"), 102],[:has(a:contains("Books")):contains("Comics"), 103],[:has(a:contains("Books")):contains("Computers &amp; Technology"), 113],[:has(a:contains("Books")):contains("Cooking"), 115],[:has(a:contains("Books")):contains("Educational"), 114],[:has(a:contains("Books")):contains("Fiction"), 107],[:has(a:contains("Books")):contains("History"), 112],[:has(a:contains("Books")):contains("Magazines"), 105],[:has(a:contains("Books")):contains("Manga"), 104],[:has(a:contains("Books")):contains("Non-fiction"), 108],[:has(a:contains("Books")):contains("Other"), 132],[:has(a:contains("Books")):contains("Religion"), 111],[:has(a:contains("Books")):contains("Sport"), 116],[:has(a:contains("Books")):contains("Textbooks"), 106],[:has(a:contains("Games")):contains("Handheld"), 92],[:has(a:contains("Games")):contains("Linux"), 87],[:has(a:contains("Games")):contains("Other"), 130],[:has(a:contains("Games")):contains("Playstation"), 97],[:has(a:contains("Games")):contains("Wii"), 91],[:has(a:contains("Games")):contains("Windows"), 85],[:has(a:contains("Games")):contains("Xbox"), 90],[:has(a:contains("Movies")):contains("3D Movies"), 69],[:has(a:contains("Movies")):contains("Animation"), 80],[:has(a:contains("Movies")):contains("Asian/Bollywood"), 79],[:has(a:contains("Movies")):contains("Blu-Ray Disc"), 148],[:has(a:contains("Movies")):contains("Cam"), 149],[:has(a:contains("Movies")):contains("Documentary"), 81],[:has(a:contains("Movies")):contains("Dubbed Movies"), 78],[:has(a:contains("Movies")):contains("DVD Disc"), 150],[:has(a:contains("Movies")):contains("HD"), 71],[:has(a:contains("Movies")):contains("Other"), 128],[:has(a:contains("Movies")):contains("Screener"), 74],[:has(a:contains("Movies")):contains("UltraHD"), 75],[:has(a:contains("Music")):contains("AAC"), 64],[:has(a:contains("Music")):contains("Karaoke"), 67],[:has(a:contains("Music")):contains("Lossless"), 23],[:has(a:contains("Music")):contains("MP3"), 22],[:has(a:contains("Music")):contains("Other"), 129],[:has(a:contains("Music")):contains("Radio Shows"), 26],[:has(a:contains("Music")):contains("Soundtrack"), 66],[:has(a:contains("Music")):contains("Transcode"), 65],[:has(a:contains("Music")):contains("Videos/Concerts"), 68],[:has(a:contains("Other")):contains("DazPoser"), 154],[:has(a:contains("Other")):contains("Other"), 138],[:has(a:contains("Other")):contains("Pictures"), 136],[:has(a:contains("Other")):contains("Subtitles"), 134],[:has(a:contains("Other")):contains("Tutorials"), 145],[:has(a:contains("Other")):contains("Wordpress"), 153],[:has(a:contains("TV")):contains("Blu-Ray Disc"), 6],[:has(a:contains("TV")):contains("Documentary"), 7],[:has(a:contains("TV")):contains("DVD Disc"), 5],[:has(a:contains("TV")):contains("HD"), 41],[:has(a:contains("TV")):contains("Other"), 151],[:has(a:contains("TV")):contains("Sport"), 146],[:has(a:contains("TV")):contains("Ultra HD"), 152],[:has(a:contains("XXX")):contains("Books"), 124],[:has(a:contains("XXX")):contains("Hentai"), 125],[:has(a:contains("XXX")):contains("Magazines"), 123],[:has(a:contains("XXX")):contains("Pictures"), 122],[:has(a:contains("XXX")):contains("UltraHD"), 121],[:has(a:contains("XXX")):contains("Videos"), 119],[:has(a:contains("XXX")):contains("XXX Games"), 126]" matched <span id="cat_12155145">
	<strong>
		<a href="torrents.php?parent_cat=Movies">Movies</a>  » DVD ISO                 </strong>
</span>
  at Jackett.Indexers.CardigannIndexer+<PerformQuery>c__async5.MoveNext () [0x0119f] in <0a574295ba6048f5b63246983ff2b96a>:0 
```

**Comment by kaso17** [comment]
looks like they renamed or added the Movies/DVD ISO category, can't test as they're currently down due to maintenance

**Comment by kaso17** [comment]
Category renaming has been fixed with 4ab0e4aa0d139d6db74637824ec768a67ebc615c
But additionally their search engine is crap. Searching "The Magnificent Seven 2016" only returns garbage.
https://katcr.co/new/torrents-search.php?search=The+Magnificent+Seven+2016&cat=0&incldead=0&freeleech=0&inclexternal=0&lang=0
Is there some way to improve the search terms?

**Comment by armani316** [comment]
@kaso17 Putting the search term between quotation marks solves the problem:

https://katcr.co/new/torrents-search.php?search=%22The+Magnificent+Seven+2016%22&cat=0&incldead=0&freeleech=0&inclexternal=0&lang=0

**Comment by kaso17** [comment]
nice catch.
Fixed with 7b31f81957d40c97eaf5a2219aa96496a9ca6b2a
